File: src/main/kotlin/com/xulog/alipay/bean/response/AlipayResponse.java
package com.xulog.alipay.bean.response;

import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.xulog.alipay.AlipayF2fPay;
import lombok.Data;

import java.io.IOException;


/**
 * Created by guofan on 2017/3/31.
 */
@Data
public class AlipayResponse<T extends BizRes> {
    private String sign = "";
    private T biz_content;

    @SuppressWarnings("unchecked")
    @JsonAnySetter
    public void setDynamicProperty(String name, Object value) {
        String writeValueAsString;
        try {
            ObjectMapper objectMapper = AlipayF2fPay.Companion.getObjectMapper();
            writeValueAsString = objectMapper.writeValueAsString(value);
            Class clazz = AlipayF2fPay.Companion.getRESPCLASSES().getOrDefault(name, BizRes.class);
            this.biz_content = objectMapper.readValue(writeValueAsString, (Class<T>) clazz);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: src/main/kotlin/com/xulog/alipay/bean/response/AlipayResponse.kt
package com.xulog.alipay.bean.response

import com.fasterxml.jackson.annotation.JsonAnySetter
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.registerKotlinModule

/**
 * Created by guofan on 2017/3/31.
 */
class AlipayResponse<T : CommonResponse>() {
    lateinit var sign: String
    lateinit var bizContent: T

    @Suppress("UNCHECKED_CAST")
    @JsonAnySetter
    fun setDynamicProperty(name: String, value: Any) {
        val objectMapper = ObjectMapper().registerKotlinModule()
        val writeValueAsString = objectMapper.writeValueAsString(value)
        bizContent = when (name) {
            "alipay_trade_precreate_response" -> objectMapper.readValue(writeValueAsString, PreCreateResponse::class.java) as T
            "alipay_trade_query_response" -> objectMapper.readValue(writeValueAsString, TradeQueryResponse::class.java) as T
            "alipay_trade_refund_response" -> objectMapper.readValue(writeValueAsString, RefundResponse::class.java) as T
            else -> {
                CommonResponse() as T
            }
        }
    }
}